MoreLinq.Test/IndexByTest.cs

```diff
@@ -0,0 +1,100 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2019 Leandro F. Vieira (leandromoh). All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class IndexByTest
+    {
+        [Test]
+        public void IndexBySimpleTest()
+        {
+            var source = new[] { "ana", "beatriz", "carla", "bob", "davi", "adriano", "angelo", "carlos" };
+            var result = source.IndexBy(x => x.First(), (item, key, index) => (item, key, index));
+
+            result.AssertSequenceEqual(
+                ( item: "ana",     key: 'a', index: 0 ),
```

Comment:
For tuples, avoid space after `(` and before `)`.

You don't need to name the elements of the tuples at all, but if it's for readability, then choose between one of the following:

- Tuples with no element names, e.g. `("ana", 'a', 0)` is acceptable (here)
- Tuples with PascalCase element names
- Anonymous-type objects with PascalCase property names (GC allocations and speed concerns are negligible for these tests)